### ISSUE TITLE
Shift key down event not being sent when shift is pressed

### ISSUE THREAD

**Issue by jhurwitz** [keyboardio/Kaleidoscope#1268]
**Describe the bug**
Using the latest firmware* on a Model 100, my computer does not see the "shift" key as being held down while I am holding it down. This is different from how my laptop's built-in keyboard acts, and how my Model 01 (using 2-year-old firmware) acted.

Some particular ways I can see this:
* I used [the Mac keyboard view](https://support.apple.com/guide/mac-help/use-the-keyboard-viewer-on-mac-mchlp1015/mac) to view what keys are being pressed, according to my computer. When pressing the shift key built-in to my laptop, I see the shift key highlighted while I'm pressing it. When pressing the shift key on my Model 100, I see nothing until I release the shift key, at which point the shift key is very briefly highlighted, so I assume both the key down and key up are sent at that time (rather than the key down being sent when the shift key is initially pressed).
* Here's a real-world example of a problem this causes. Visit any file in Github, like https://github.com/keyboardio/Kaleidoscope/blob/master/src/Kaleidoscope.h. Try to hold down the shift key in order to select multiple lines of code. This is possible using my laptop's built-in keyboard, but not possible using my Model 100 because pressing shift (with no other keys) doesn't send shift to my computer.

(*Slightly modified from the default firmware though; I'm on a different computer right now and cannot easily flash back to the default firmware to try to reproduce this bug. If others cannot repro, let me know and I can see if my firmware is at fault.)
 
**To Reproduce**
See above

**Expected behavior**
I expect a shift key down event to be sent when shift is initially pressed.

**Environment (please complete the following information):**
 - macOS 12.6
 - Keyboardio Model 100

**Additional context**

**Comment by jhurwitz** [comment]
> I've seen this problem before, and in most (all?) cases it turned out that SpaceCadet mode was somehow enabled, usually not intentionally. Can you fire up Chrysalis, go to Preferences, My Keyboard, and see if SpaceCadet is on? If it is, turn it off, and see if that helps.

Maybe? I'm using a completely custom firmware (no Chrysalis) so I just downloaded and opened Chrysalis for the first time. I got a warning that "Chrysalis no longer supports configurations containing a mix of hardcoded and EEPROM layers." and I am unable to use most of Chrysalis, but I was able to open preferences and saw (to my surprise) that SpaceCadet was enabled. I toggled it off.

One of those steps in the process though (not sure which) seems to have bricked my keyboard. The LEDs are still on, but none of the keys seem to do anything now.
